app/models/document.rb

```diff
@@ -37,4 +37,12 @@ class Document < ApplicationRecord
     'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet', # .xlsx
     'text/plain'
   ]
+  def self.join_documents_to_filenames
+    joins(<<-SQL)
+      INNER JOIN active_storage_attachments
+      ON active_storage_attachments.record_id = documents.id
+      INNER JOIN active_storage_blobs
+      ON active_storage_blobs.id = active_storage_attachments.blob_id
+    SQL
+  end
```

Comment:
* Use `#scope` to set up this (check the rails docs)
* I think you can use `#joins` here instead of manually building the SQL string. E.g.: `joins(active_storage_attachments: [:active_storage_blobs])`
* this join might not be necessary. Try something like this : https://medium.com/fiatinsight/how-to-change-a-filename-in-rails-active-storage-f3e4f26f427e
* -> `join_filenames_to_documents`